kafka-client/src/main/java/dev/responsive/kafka/internal/db/partitioning/SegmentPartitioner.java

```diff
@@ -103,6 +103,30 @@ public SegmentPartition(final int tablePartition, final long segmentId) {
       this.segmentId = segmentId;
     }
 
+    @Override
```

Comment:
I know this isn't part of this PR, but how hard would it be to use the segment start timestamp for the segment ID, rather than the index? It would help some when browsing the database. Right now it's hard to make sense of what stream times the segments correspond to.

Reply:
FYI I changed it to use the segment start timestamp in this PR, for the collection name only. Honestly, if we do this then imo we should just go ahead and convert all instances of the segment id to "segment start timestamp" instead, and get rid of the "segment id" concept entirely

That feels reasonable but I'll save that for a followup PR. Filed a ticket: [here](https://www.notion.so/responsivedev/Replace-segment-id-with-start-timestamp-3582ca33324148169cca8d1d04b870de?pvs=4)

Reply:
sounds good. thanks!